src/storage/data_table.cpp

```diff
@@ -589,7 +651,13 @@ void DataTable::AddTileGroupWithOidForRecovery(const oid_t &tile_group_id) {
 
 }
 
+// NOTE: This function is only used in test cases.
```

Comment:
I also noted this problem. I tried to remove this function but found its hard to do that, as many test cases invoke this function, especially those testing the correctness of tile groups.